### PR TITLE
Exit castle by [X] button

### LINKED ISSUE #649: Cannot close game while battle in castle
While defending a castle player cannot cancel fight: our version accepts only retreat from the battle, but player cannot retreat from a castle.

### PR DESCRIPTION
**Pull request by vasilenkoalexey**
Partially fix #649

### CODE REVIEW

**Comment by ihhub** [comment]
Hi @vasilenkoalexey , the fix does nothing :)

**Comment by vasilenkoalexey** [comment]
> Hi @vasilenkoalexey , the fix does nothing :)

You right, my bad... But it pr allow to exit from castle by [X] button...

File: src/fheroes2/castle/castle_dialog.cpp [hunk 1]
```diff
@@ -319,7 +319,7 @@ int Castle::OpenDialog( bool readonly )
     fheroes2::Image surfaceHero( 552, 105 );
 
     // dialog menu loop
-    while ( le.HandleEvents() ) {
+    while ( le.HandleEvents( true, true ) ) {
```
Comment (ihhub) [inline src/fheroes2/castle/castle_dialog.cpp:322]: This change just closes castle's window while the issue state to close the game showing a confirmation dialog for this.

**Review by ihhub — CHANGES_REQUESTED** [review]

**Comment by vasilenkoalexey** [comment]
So, I think to completely fix # 649 we need a more complex solution

**Pull request closed without merging**